src/mlpack/methods/preprocess/preprocess_split_main.cpp

```diff
@@ -76,10 +76,13 @@ static void mlpackMain()
   // Parse command line options.
   const double testRatio = CLI::GetParam<double>("test_ratio");
 
-  if (CLI::GetParam<int>("seed") == 0)
-    mlpack::math::RandomSeed(std::time(NULL));
-  else
-    mlpack::math::RandomSeed((size_t) CLI::GetParam<int>("seed"));
+#if(BINDING_TYPE != BINDING_TYPE_TEST) // This is a not unit test
```

Comment:
I don't think you handled the second comment yet:

> If we perform the check here, inside the bindings, we have to modify every single binding that has a seed option. Wouldn't it be better to make changes to the RandomSeed() function directly?

Reply:
I'm testing that. Please give me some time. I'll handle all the requirements after my exam. I'm extremely sorry for the delay :(

Reply:
Sure, no worries.  There is no hurry at all and no reason to apologize. :)  I thought this was ready since you pushed some commits.  Good luck with your exam!